app/controllers/api/metric_visibilities_controller.rb

```diff
@@ -10,33 +10,56 @@ class Api::MetricVisibilitiesController < FrontendController
   def toggle_visible
     @metric.toggle_visible_for_plan(@plan)
 
+    errors = @metric.errors.presence
+    if errors
+      flash[:error] = errors.full_messages.to_sentence
+    else
+      metric = @metric.method_metric? ? 'Method' : 'Metric'
```

Comment:
The amount of this snippet repeating throught the code (not only in this file) is crazy. I don't get why it's not a method on the `Metric` model :shrug: 

But leave it for another PR...